dev/PushNotifications/PushNotificationReceivedEventArgs.cpp

```diff
@@ -60,51 +49,45 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         return { payload, payload + (length * sizeof(uint8_t)) };
     }
 
-    std::vector<uint8_t> PushNotificationReceivedEventArgs::BuildPayload(winrt::hstring const& payload)
-    {
-        std::string payloadToSimpleString{ ::winrt::Microsoft::Windows::PushNotifications::Helpers::WideStringToUtf8String(payload) };
-        return { payloadToSimpleString.c_str(), payloadToSimpleString.c_str() + (payloadToSimpleString.length() * sizeof(uint8_t)) };
-    }
-
     winrt::com_array<uint8_t> PushNotificationReceivedEventArgs::Payload()
     {
         return { m_rawNotificationPayload.data(), m_rawNotificationPayload.data() + (m_rawNotificationPayload.size() * sizeof(uint8_t)) };
     }
 
     winrt::BackgroundTaskDeferral PushNotificationReceivedEventArgs::GetDeferral()
     {
-        if (!m_unpackagedAppScenario)
+        if (!PushNotificationHelpers::IsPackagedAppScenario())
         {
-            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
-
-            return m_backgroundTaskInstance.GetDeferral();
-        }
-        else
-        {
-            auto dummyDeferral = winrt::make<PushNotificationDummyDeferral>();
-            return dummyDeferral.as<winrt::BackgroundTaskDeferral>();
+            return winrt::make<PushNotificationDummyDeferral>().as<BackgroundTaskDeferral>();
         }
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, m_isForegroundActivation, "Foreground activation cannot call this.");
```

Comment:
So why is this check not at the beginning of the function? Is that on purpose?

Reply:
Ok so it sounds like this should be a check that is at the beginning of the function and applicable to both packaged/unpackaged scenarios? Or am i wrong?

Reply:
Yup it should be, but since we have decided to noop from beginning, it would change the behavior of the API and that needs further review.